patch/attiny88.yaml

```diff
@@ -0,0 +1,127 @@
+        PRESCALER_8: [3, "Prescaler Value 8"]
+        PRESCALER_16: [4, "Prescaler Value 16"]
+        PRESCALER_32: [5, "Prescaler Value 32"]
+        PRESCALER_64: [6, "Prescaler Value 64"]
+        PRESCALER_128: [7, "Prescaler Value 128"]
+  ADCSRB:
+    ADTS:
+      _replace_enum:
+        FREE: [0, "Free Running mode"]
+        AC: [1, "Analog Comparator"]
+        INT0: [2, "External Interrupt Request 0"]
+        TC0_CMA: [3, "Timer/Counter0 Compare Match A"]
+        TC0_OVF: [4, "Timer/Counter0 Overflow"]
+        TC1_CMB: [5, "Timer/Counter1 Compare Match B"]
+        TC1_OVF: [6, "Timer/Counter1 Overflow"]
+        TC1_CE: [7, "Timer/Counter1 Capture Event"]
+CPU:
+  _modify:
+    CLKPR:
+      access: read-write
+    OSCCAL:
+      access: read-write
+    PORTCR:
+      access: read-write
+    PRR:
+      access: read-write
+  CLKPR:
+    CLKPS:
+      _replace_enum:
+        PRESCALER_1: [0, "Prescaler Value 1"]
+        PRESCALER_2: [1, "Prescaler Value 2"]
+        PRESCALER_4: [2, "Prescaler Value 4"]
+        PRESCALER_8: [3, "Prescaler Value 8"]
+        PRESCALER_16: [4, "Prescaler Value 16"]
+        PRESCALER_32: [5, "Prescaler Value 32"]
+        PRESCALER_64: [6, "Prescaler Value 64"]
+        PRESCALER_128: [7, "Prescaler Value 128"]
+        PRESCALER_256: [8, "Prescaler Value 256"]
+  SMCR:
+    SM:
+      _replace_enum:
+        IDLE: [0, "Idle"]
+        ADC: [1, "ADC Noise Reduction"]
+        PDOWN: [2, "Power-down"]
+EEPROM:
+  EECR:
+    EEPM:
+      _replace_enum:
+        ATOMIC: [0, "Atomic (erase and write in one operation)"]
+        ERASE: [1, "Erase only"]
+        WRITE: [2, "Write only"]
+EXINT:
+  EICRA:
+    ISC0:
+      _replace_enum:
+        LOW: [0, "The low level of INT0 generates an interrupt request"]
+        TOGGLE: [1, "Any logical change on INT0 generates an interrupt request"]
+        FALLING: [2, "The falling edge of INT0 generates an interrupt request"]
+        RISING: [3, "The rising edge of INT0 generates an interrupt request"]
+    ISC1:
+      _replace_enum:
+        LOW: [0, "The low level of INT1 generates an interrupt request"]
+        TOGGLE: [1, "Any logical change on INT1 generates an interrupt request"]
+        FALLING: [2, "The falling edge of INT1 generates an interrupt request"]
+        RISING: [3, "The rising edge of INT1 generates an interrupt request"]
+  _modify:
+    PCICR:
+      description: "Pin Change Interrupt Control Register"
+PORTA:
+  _modify:
+    PORTA:
+      access: read-write
+    DDRA:
+      access: read-write
+TWI:
+  TWCR:
+    _modify:
+      TWWC:
+        access: read-only
+  TWAMR:
+    _modify:
+      TWAM:
+        description: "TWI (Slave) Address Mask Bits"
+  TWSR:
+    _modify:
+      TWS:
+        access: read-only
+    TWPS:
+      _replace_enum:
+        PRESCALER_1: [0, "Prescaler Value 1"]
+        PRESCALER_4: [1, "Prescaler Value 4"]
+        PRESCALER_16: [2, "Prescaler Value 16"]
+        PRESCALER_64: [3, "Prescaler Value 64"]
```

Comment:
Was there a reason you couldn't use the common TWI patch?  Maybe we can change the patch in some way to make it compatible?

Reply:
The common patch chokes on 
```yaml
TWI:
  _modify:
    TWCR:
      access: read-write
```
with
```
Traceback (most recent call last):
  File "patch/svdpatch.py", line 834, in <module>
    main()
  File "patch/svdpatch.py", line 827, in main
    process_device(svd, root)
  File "patch/svdpatch.py", line 806, in process_device
    update_fields)
  File "patch/svdpatch.py", line 732, in process_peripheral
    process_peripheral_modify(ptag, rspec, rmod)
  File "patch/svdpatch.py", line 218, in process_peripheral_modify
    rtag.find(key).text = value
AttributeError: 'NoneType' object has no attribute 'text'
Makefile:26: recipe for target 'svd/attiny88.patched.svd' failed
make: *** [svd/attiny88.patched.svd] Error 1
```

It would seem that this particular model's ATDF doesn't specify read-only for the control register.

Reply:
Ok, I got this one fixed by looking back at your changes in e0d031a3 and adapting them.

Reply:
Ah, right, I was thinking about this while preparing the svdpatch changes for upstream.  Looks like it came back to bite earlier than I thought!